src/settingsui.ts

```diff
@@ -344,9 +401,12 @@ class SettingElem {
 				} else if (!value) {
 					if (this.props.instant && typeof userscript.unload === 'function') {
 						userscript.unload();
+						userscript.hasRan = false;
```

Comment:
i'm not sure about the behaviour of hasRan in relation to how the metadata is handled. i think i didn't turn it off because once a script with metadata has ran, crankshaft knows it's metadata and keeps them even if it get's unloaded. i might be wrong so i'll check the hasRan usage